contracts/nexus-gateway/src/contract.rs

```diff
@@ -115,80 +111,68 @@ mod tests {
     const AXELARNET_GATEWAY: &str = "axelarnet_gateway";
 
     #[test]
-    fn call_contract_with_token_no_token() {
+    fn route_message_with_token_unauthorized() {
         let mut deps = mock_dependencies();
         instantiate_contract(deps.as_mut());
 
         let res = execute(
             deps.as_mut(),
             mock_env(),
-            mock_info(NEXUS, &[]),
-            ExecuteMsg::CallContractWithToken {
-                destination_chain: "destinationChain".parse().unwrap(),
-                destination_address: "0xD419".parse().unwrap(),
-                payload: decode("bb9b5566c2f4876863333e481f4698350154259ffe6226e283b16ce18a64bcf1")
-                    .unwrap()
-                    .into(),
-            },
+            mock_info("unauthorized", &[Coin::new(100, "test")]),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
         );
-        assert!(res.is_err_and(|err| err_contains!(err.report, Error, Error::InvalidToken { .. })));
+        assert!(res.is_err_and(|err| err_contains!(
+            err.report,
+            permission_control::Error,
+            permission_control::Error::AddressNotWhitelisted { .. }
+        )));
     }
 
     #[test]
-    fn call_contract_with_token() {
-        let mut tx_hash = [0u8; 32];
-        rand::thread_rng().fill_bytes(&mut tx_hash);
-        let nonce = rand::random();
+    fn route_message_with_token_invalid_token() {
+        let mut deps = mock_dependencies();
+        instantiate_contract(deps.as_mut());
+
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info(AXELARNET_GATEWAY, &[]),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
+        );
+        assert!(res.is_err_and(|err| { err.to_string().contains("invalid token") }));
 
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info(
+                AXELARNET_GATEWAY,
+                &[Coin::new(100, "test"), Coin::new(100, "test")],
+            ),
+            ExecuteMsg::RouteMessageWithToken(router_messages()[0].clone()),
+        );
+        assert!(res.is_err_and(|err| { err.to_string().contains("invalid token") }));
+    }
+
+    #[test]
+    fn route_message_with_token() {
```

Comment:
Can we start moving new integration tests to the tests folder to follow the new pattern used in axelarnet-gateway?

Reply:
I am not sure if I like that pattern

Reply:
we should discuss how we want to do this in the sync for consistency